typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
Here's a quick sketch of what I'm talking about. I don't think you need to use `super` if you use this pattern: https://gist.github.com/maxsond/bee981a3e0d80a60e591b6faf0c5c3ba